packages/dd-trace/src/format.js

```diff
@@ -64,6 +68,33 @@ function setSingleSpanIngestionTags (span, options) {
   addTag({}, span.metrics, SPAN_SAMPLING_MAX_PER_SECOND, options.maxPerSecond)
 }
 
+// TODO: do we need to handle both otel and dd trace span links in this function or only dd trace span links?
+function extractSpanLinks (trace, span) {
+  const links = []
+  if (span._links) {
+    for (const link of span._links) {
+      const { context, attributes } = link
+      const linksArrayString = links.join()
+
+      const formattedLink = formatLink(span.context(), context, attributes)
+      let formattedLinkString = spanLinkToString(formattedLink)
+
+      if ((Buffer.byteLength(linksArrayString) + spanLinkLength(formattedLink, formattedLinkString)) >=
+      MAX_SPAN_LINKS_LENGTH) {
+        formattedLink.dropped_attributes_count += formattedLink.attributesCount
+        formattedLink.attributesCount = 0
+      }
+      // Update the string after possible flushing
+      formattedLinkString = spanLinkToString(formattedLink)
+      if ((Buffer.byteLength(linksArrayString) + spanLinkLength(formattedLink, formattedLinkString)) <
```

Comment:
No I don't believe it's guaranteed, because even after flushing the size might >= limit and if that's the case then we just don't add the link at all. This is supposed to follow the algorithm in the RFC